include/seqan/blast/blast_statistics.h

```diff
@@ -0,0 +1,1076 @@
+ * Blast (and many other tools) compute scores of a stretch of gaps as
+ * <tt>s = gO + n * gE</tt>
+ * where gO is the gapOpen score, gE is the gap extend score and n ist the
+ * total number of gap characters.
+ *
+ * SeqAn, however, normally computes them as as <tt>s = gO + (n-1) * gE</tt>.
+ *
+ * Effectively this results only in a different gap open score. To make dealing with this easier, there are
+ * convenience functions for setting and getting the gap open score in blast convention:
+ * @link BlastScoringScheme#scoreGapOpenBlast @endlink and @link BlastScoringScheme#setScoreGapOpenBlast @endlink.
+ *
+ * @section Karlin-Altschul values
+ *
+ * E-Value statistics require certain pre-computed constants. The BlastScoringScheme picks these automatically
+ * whenever you modify it. Unfortunately there aren't parameters for all combinations of scoring schemes and gap
+ * costs, so you have to check @link BlastScoringScheme#isValid @endlink after modifying it or verify the return
+ * value of the set* functions which is bool here (instead of void).
+ *
+ * More details on the scoring parameters is in these slides:
+ * <a href="https://www.cs.umd.edu/class/fall2011/cmsc858s/Local_Alignment_Statistics.pdf">
+ * https://www.cs.umd.edu/class/fall2011/cmsc858s/Local_Alignment_Statistics.pdf</a>.
+ *
+ * The constants used in SeqAn were imported from the NCBI blast source code (blast-2.2.26)
+ *
+ * @tparam TScoringScheme A SeqAn Score type
+ */
+
+
+/* NOTE(h-2): Maybe this should specialize Score<> instead of being a wrapper.
+ * This would however involve adding a new template parameter to score and
+ * changing lots of code. Also it would become confusing as the shortcuts
+ * like Blosum62 would no longer work...
+ */
+template <typename TScore>
+struct BlastScoringScheme
+{
+    typedef typename Value<TScore>::Type TValue;
+    typedef typename KarlinAltschulValues<TScore, void>::size_t TNumValues;
+
+    /* related to ScoringScheme conversion */
+    TValue _gapOpenBlast = 0;
+    TScore _internalScheme;
+
+    /* parameter selection */
+    TNumValues parameterIndex = std::numeric_limits<TNumValues>::max();
+};
+
+// ============================================================================
+// Functions
+// ============================================================================
+
+/*!
+ * @fn BlastScoringScheme#seqanScheme
+ * @headerfile <seqan/blast.h>
+ * @brief Retrieve a @link Score @endlink object for use with e.g. @link localAlignment @endlink.
+ * @signature TScoringScheme const & seqanScheme(blastScoringScheme);
+ * @return TScoringScheme_const_& the internal scoring object for use with other SeqAn functions.
+ *
+ * @section Remarks
+ *
+ * The reference returned is always const, because you should not modify the internal score object
+ * directly. Use @link BlastScoringScheme @endlink's interface instead.
+ */
+
+template <typename TScore>
+inline TScore const &
+seqanScheme(BlastScoringScheme<TScore> const & bscheme)
```

Comment:
I am using the term BlastScoringScheme and SeqanScoringScheme to differentiate between the behaviours, so it has to be clear that this function returns an object with SeqAn behaviour.